apps/distgo/cmd/clean/clean.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clean
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+
+	"github.com/palantir/godel/apps/distgo/params"
+)
+
+const (
+	defaultBuildOutputDir = "build"
+	defaultDistOutputDir  = "dist"
+)
+
+func Clean(products []string, cfg params.Project) error {
+	// Only remove default output directories if no products specified
+	if len(products) == 0 {
+		if cfg.BuildOutputDir == "" {
+			cfg.BuildOutputDir = defaultBuildOutputDir
+		}
+		if err := remove(cfg.BuildOutputDir); err != nil {
+			return errors.WithStack(err)
+		}
+
+		if cfg.DistOutputDir == "" {
+			cfg.DistOutputDir = defaultDistOutputDir
+		}
+		if err := remove(cfg.DistOutputDir); err != nil {
+			return errors.WithStack(err)
+		}
```

Comment:
Should there be a `return nil` at the end of this `if` block? As currently written, if `len(products) == 0`, this will remove both default directories and all project directories. Is that intended?